File: backend/app/schemas/imports.py
import uuid
from typing import Literal

from pydantic import BaseModel, ConfigDict


class ImportErrorResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    row_number: int
    field_name: str | None
    error_code: str
    message: str
    raw_value: str | None


class ImportBatchResponse(BaseModel):
    id: uuid.UUID
    filename: str
    status: str
    total_rows: int
    valid_rows: int
    error_rows: int


class ImportPreviewResponse(ImportBatchResponse):
    rows: list[dict]
    errors: list[ImportErrorResponse]


class ConfirmImportRequest(BaseModel):
    mode: Literal["upsert", "create_only", "update_only"] = "upsert"


class ImportResultResponse(ImportBatchResponse):
    created: int
    updated: int
    skipped: int

File: frontend/src/types/imports.ts
export type ImportError = {
  id: string;
  row_number: number;
  field_name: string | null;
  error_code: string;
  message: string;
  raw_value: string | null;
};

export type ImportBatch = {
  id: string;
  filename: string;
  status: string;
  total_rows: number;
  valid_rows: number;
  error_rows: number;
};

export type ImportPreview = ImportBatch & {
  rows: Array<Record<string, string>>;
  errors: ImportError[];
};

export type ImportResult = ImportBatch & {
  created: number;
  updated: number;
  skipped: number;
};
